crates/common/src/provider/alloy.rs

```diff
@@ -250,10 +255,10 @@ impl ProviderBuilder {
 ///   - polygon
 ///
 /// Fallback is the default [`Provider::estimate_eip1559_fees`] implementation
-pub async fn estimate_eip1559_fees<P: TempProvider>(
+pub async fn estimate_eip1559_fees<P: Provider<Ethereum, T>, T: Transport + Clone>(
```

Comment:
I think we could solve this with a layer as well, but can simplify it separately.